I found a small regression when running the london harvesters from a clear start:

```
Traceback (most recent call last):
  File "/usr/bin/ckan", line 8, in <module>
    sys.exit(ckan())
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/usr/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/srv/app/src/ckanext-harvest/ckanext/harvest/cli.py", line 284, in run_test
    utils.run_test_harvester(id, force_import)
  File "/srv/app/src/ckanext-harvest/ckanext/harvest/utils.py", line 411, in run_test_harvester
    lib.run_harvest_job(job_obj, harvester)
  File "/srv/app/src/ckanext-harvest/ckanext/harvest/tests/lib.py", line 36, in run_harvest_job
    obj_ids = queue.gather_stage(harvester, job)
  File "/srv/app/src/ckanext-harvest/ckanext/harvest/queue.py", line 432, in gather_stage
    harvest_object_ids = harvester.gather_stage(job)
  File "/srv/app/src_extensions/ckanext-datapress_harvester/ckanext/datapress_harvester/harvesters/datapress.py", line 234, in gather_stage
    pkg_dicts = self._fetch_packages(remote_datapress_base_url)
  File "/srv/app/src_extensions/ckanext-datapress_harvester/ckanext/datapress_harvester/harvesters/datapress.py", line 403, in _fetch_packages
    extra_resource_fields = (list(map(normalise_ckan_resources, extra_resource_fields)))
  File "/srv/app/src_extensions/ckanext-datapress_harvester/ckanext/datapress_harvester/harvesters/datapress.py", line 29, in normalise_ckan_resources
    normalised_resources = package_dict.get('resources',[])
AttributeError: 'str' object has no attribute 'get'
```

But have [fixed up the commit with an appropriate patch](https://github.com/GreaterLondonAuthority/dfl-ckanext-havester/pull/21/commits/d7b2d36a0b3cd319d3fc470b98de7fa39a9d7ac8#diff-06aa463e00150690c24a473b90efb7eb10ee799c1816e65620559f024266b155R404-R411) to make it work.

--- a/ckanext/datapress_harvester/harvesters/datapress.py
+++ b/ckanext/datapress_harvester/harvesters/datapress.py
@@ -23,7 +23,7 @@ from ckanext.harvest.model import HarvestObject
 log = logging.getLogger(__name__)
 
 EXTRA_PKG_FIELDS = ['london_smallest_geography', 'update_frequency']
-EXTRA_RESOURCE_FIELDS = ['temporal_coverage_from', 'temporal_coverage_to']
+EXTRA_RESOURCE_FIELDS = ['temporal_coverage_from', 'temporal_coverage_to', 'url']
 
 def normalise_ckan_resources(package_dict):
     normalised_resources = package_dict.get('resources',[])
@@ -350,8 +350,8 @@ class DataPressHarvester(HarvesterBase, DFLHarvesterMixin):
                    
                    normalised_resources.append({res_id: res_obj})
                package_dict['resources'] = normalised_resources
+               lookup[pkg_id] = package_dict
 
-            
             if isinstance(package_dict.get('resources'), dict):
                for res_id, res_obj in package_dict.get('resources',[]).items():                   
                    resource_extra_fields = {}
@@ -400,12 +400,25 @@ class DataPressHarvester(HarvesterBase, DFLHarvesterMixin):
         for dataset_dict in results:
             extra_fields = self.extra_fields_lookup.get(dataset_dict["id"], {})
             extra_resource_fields = extra_fields.pop('resources',[])
-            for resource_obj in dataset_dict.get('resources',[]):
-                res_id = resource_obj['id']
-                if res_id in extra_resource_fields:
-                    resource_obj.update(extra_resource_fields[res_id])
 
-            dataset_dict.update(extra_fields)
+            # Convert extra_resource_fields to a dictionary for faster lookup by res_id
+            if isinstance(extra_resource_fields,dict):
+                # london datapress format is already a dict of id -> resource
+                extra_resource_fields_dict = extra_resource_fields
+            else:
+                # brent/barnet datapress format as a list of resources
+                # (which have an id) so convert it into london format
+                extra_resource_fields_dict = {res_id: res_obj for resource in extra_resource_fields for res_id, res_obj in resource.items()}
+
+
+            resources = dataset_dict.get('resources') or dataset_dict.get('organization', {}).get('resources', [])
+
+            for resource_obj in resources:
+                    res_id = resource_obj['id']
+                    if res_id in extra_resource_fields_dict:
+                        resource_obj.update(extra_resource_fields_dict[res_id])
+
+            dataset_dict.update(extra_fields)  
         
         return results
 
@@ -502,7 +515,7 @@ class DataPressHarvester(HarvesterBase, DFLHarvesterMixin):
             # tweaks like this on a per-datapress-instance basis.
             # Do all DataPress instances work similarly? Can we use the harvest
             # URL here?
-            if resource["url"].startswith("https://airdrive-secure.s3-eu-west-1"):
+            if "url" in resource and resource["url"].startswith("https://airdrive-secure.s3-eu-west-1"):
                 base = "https://data.london.gov.uk/download"
                 dataset = package_dict["name"]
                 id = resource["id"]
@@ -510,11 +523,12 @@ class DataPressHarvester(HarvesterBase, DFLHarvesterMixin):
                 format = resource["format"]
                 resource["url"] = f"{base}/{dataset}/{id}/{file}.{format}"
 
-            if "format" not in resource or not resource["format"]:
-                resource["format"] = self._resource_format_from_url(resource["url"])
+            if "url" in resource:
+                if "format" not in resource or not resource["format"]:
+                    resource["format"] = self._resource_format_from_url(resource["url"])
 
-            if resource["format"] == "image":
-                resource["format"] = self._guess_image_format(resource["url"])
+                if resource["format"] == "image":
+                    resource["format"] = self._guess_image_format(resource["url"])
 
         # Remove the timezone from the dates. CKAN doesn't store it internally and it
         # messes up date-based comparisons later if the timezone is kept (because the base
@@ -567,6 +581,8 @@ class DataPressHarvester(HarvesterBase, DFLHarvesterMixin):
                     base_context.copy(), package_dict
                 )
                 return True
+
+            normalise_ckan_resources(package_dict)
 
             package_dict = self._datapress_to_ckan(package_dict, harvest_object)
 
@@ -747,8 +763,6 @@ class DataPressHarvester(HarvesterBase, DFLHarvesterMixin):
                 },
             ]
 
-            normalise_ckan_resources(package_dict)
-            
             for resource in package_dict.get("resources", []):
                 
                 # Clear remote url_type for resources (eg datastore, upload) as

--- a/ckanext/datapress_harvester/harvesters/datapress.py
+++ b/ckanext/datapress_harvester/harvesters/datapress.py
@@ -23,7 +23,7 @@ from ckanext.harvest.model import HarvestObject
 log = logging.getLogger(__name__)
 
 EXTRA_PKG_FIELDS = ['london_smallest_geography', 'update_frequency']
-EXTRA_RESOURCE_FIELDS = ['temporal_coverage_from', 'temporal_coverage_to', 'url']
+EXTRA_RESOURCE_FIELDS = ['temporal_coverage_from', 'temporal_coverage_to', 'url', 'timeFrame']
 
 def normalise_ckan_resources(package_dict):
     normalised_resources = package_dict.get('resources',[])
@@ -529,6 +529,12 @@ class DataPressHarvester(HarvesterBase, DFLHarvesterMixin):
 
                 if resource["format"] == "image":
                     resource["format"] = self._guess_image_format(resource["url"])
+
+            if "timeFrameFrom" in resource:
+                resource["temporal_coverage_from"] = datetime.strptime(resource["timeFrameFrom"], "%Y-%m").strftime("%Y-%m-%d")
+
+            if "timeFrameTo" in resource:
+                resource["temporal_coverage_to"] = datetime.strptime(resource["timeFrameTo"], "%Y-%m").strftime("%Y-%m-%d")
 
         # Remove the timezone from the dates. CKAN doesn't store it internally and it
         # messes up date-based comparisons later if the timezone is kept (because the base

